feat(sqla_factory): added __set_association_proxy__ attribute

nisemenov opened the pull request:
<!--
By submitting this pull request, you agree to:
- follow [Litestar's Code of Conduct](https://github.com/litestar-org/.github/blob/main/CODE_OF_CONDUCT.md)
- follow [Litestar's contribution guidelines](https://github.com/litestar-org/.github/blob/main/CONTRIBUTING.md)
- follow the [PSFs's Code of Conduct](https://www.python.org/psf/conduct/)
-->
## Description

- **\_\_set_association_proxy\_\_** attribute has been added in SQLAlchemyFactory class to add AssociationProxy type fields in the result FieldMeta factory dict;
- appropriate tests have been added;
- Usage Guide Configuration section has been updated.

<!--
Please add in issue numbers this pull request will close, if applicable
Examples: Fixes #4321 or Closes #1234

Ensure you are using a supported keyword to properly link an issue:
https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->
## Closes
- Closes #624

adhtruong commented on docs/usage/library_factories/sqlalchemy_factory.rst:
```diff
@@ -10,21 +10,39 @@ Basic usage is like other factories
+++++++++++++
 
-By default, relationships will not be set. This can be overridden via ``__set_relationships__``.
+By default, ``__set_relationships__`` is set to ``False``. If it is ``True``, all fields with the SQLAlchemy `relationship() <relationship()_>`_ will be included in the resulting mock dictionary created by ``build`` method.
```
```suggestion
By default, ``__set_relationships__`` is set to ``False``. If it is ``True``, all fields with the SQLAlchemy `relationship() <relationship()_>`_ will be included in the result created by ``build`` method.
```

adhtruong commented on docs/usage/library_factories/sqlalchemy_factory.rst:
```diff
@@ -10,21 +10,39 @@ Basic usage is like other factories
 
 .. note::
-    In general, foreign keys are not automatically generated by ``.build``. This can be resolved by setting the fields yourself and/or using ``create_sync``/ ``create_async`` so models can be added to a SQLA session so these are set.
+    In general, ForeignKey fields are automatically generated by ``build`` method because :class:`__set_foreign_keys__ <polyfactory.factories.sqlalchemy_factory.SQLAlchemyFactory.__set_foreign_keys__>` is set to ``True`` by default. But their values can be overwritten by using ``create_sync``/ ``create_async`` methods, so SQLAlchemy ORM creates them.
```
Is the last sentence correct always? Would that only be the case if a relationship is present?

nisemenov commented:
Hi, @adhtruong! 
Thanks for reviewing. I have fixed all issues.

More detailed about https://github.com/litestar-org/polyfactory/pull/629#discussion_r1919113919: I have rewritten this part more clearly.

adhtruong commented:
Thanks for updating! LGTM

adhtruong commented:
@all-contributors add @nisemenov for code doc
